packages/transaction-controller/src/utils/batch.ts

```diff
@@ -665,3 +642,61 @@ function wipeTransactionBatchById(
     );
   });
 }
+
+/**
+ * Prepares the approval data for a transaction batch.
+ *
+ * @param options - The options object containing necessary parameters.
+ * @param options.batchId - The batch ID for the transaction batch.
+ * @param options.chainId - The chain ID of the transactions.
+ * @param options.from - The sender's address.
+ * @param options.isSimulationEnabled - A function to check if simulation is enabled.
+ * @param options.nestedTransactions - The array of nested transactions.
+ * @param options.networkClientId - The network client ID.
+ * @param options.origin - The origin of the transaction batch.
+ * @param options.update - The update function to modify the transaction controller state.
+ * @returns The prepared transaction batch metadata.
+ */
+async function prepareApprovalData({
+  batchId,
+  chainId,
+  from,
+  isSimulationEnabled,
+  nestedTransactions,
+  networkClientId,
+  origin,
+  update,
+}: {
+  batchId: Hex;
+  chainId: Hex;
+  from: Hex;
+  isSimulationEnabled: () => boolean;
+  nestedTransactions: TransactionBatchSingleRequest[];
+  networkClientId: string;
+  origin?: string;
+  update: UpdateStateCallback;
+}): Promise<TransactionBatchMeta> {
+  if (!isSimulationEnabled()) {
+    throw new Error('Simulation is not enabled');
```

Comment:
```suggestion
    throw new Error('Cannot create transaction batch as simulation not supported');
```

Reply:
This is something we'll need to consider in any client features that use a transaction batch, as we currently require simulation.

In future we could skip simulation if gas limits are provided on each transaction, or we could just estimate as we go normally and display no gas fee.